### PR TITLE
ADM-471: [frontend] Error handling enhancement for frontend

### PR DESCRIPTION
**Pull request by TingLiMio**
## Summary

- Handle `204` for get pipeline steps.
- Fix no done cards warning cannot reset issue. 

**Before**

1. Select a time range without a done card.
2. A warning display like `Sorry there is no card has been done, please change your collection date!` and Board Verify failed.
3. Then select a time range with done cards.
4. A warning still display, but Board verify succeed.

Screenshot 2023-07-10 at 16 48 32

**After**
1. Select a time range without a done card.
2. A warning display like `Sorry there is no card has been done, please change your collection date!` and Board Verify failed.
3. Then select a time range with done cards.
4. A warning not display, and Board verify succeed.

Screenshot 2023-07-10 at 16 50 30

### CODE REVIEW

File: frontend/__tests__/src/client/MetricsClient.test.ts [hunk 1]
```diff
@@ -47,4 +49,12 @@ describe('get steps from metrics response', () => {
       await metricsClient.getSteps(params, buildId, organizationId, pipelineType, token)
     }).rejects.toThrow(VERIFY_ERROR_MESSAGE.BAD_REQUEST)
   })
+
+  it('should show isNoStep True when getSteps response status 204', async () => {
```
Comment (guzhongren) [inline frontend/__tests__/src/client/MetricsClient.test.ts:53]: 1. should set isNoStep to true when ....
maybe this is better than yours

2. haveStep is better than isNoStep

File: frontend/src/clients/board/BoardClient.ts [hunk 2]
```diff
@@ -8,6 +8,9 @@ export class BoardClient extends HttpClient {
   response = {}
 
   getVerifyBoard = async (params: BoardRequestDTO) => {
+    this.isBoardVerify = false
+    this.isNoDoneCard = false
```
Comment (guzhongren) [inline frontend/src/clients/board/BoardClient.ts:12]: `isNoDoneCard`, maybe using `haveDoneCard` is good than yours.

**Review by guzhongren — COMMENTED** [review]
Need to resolve my comments

File: frontend/__tests__/src/client/MetricsClient.test.ts (hunk 1 above)
Comment (TingLiMio) [reply frontend/__tests__/src/client/MetricsClient.test.ts:53]: Done

File: frontend/src/clients/board/BoardClient.ts (hunk 2 above)
Comment (TingLiMio) [reply frontend/src/clients/board/BoardClient.ts]: Done

**Review by guzhongren — APPROVED** [review]

**Pull request merged**